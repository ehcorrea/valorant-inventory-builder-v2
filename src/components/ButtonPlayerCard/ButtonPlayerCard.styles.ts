import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import { Image, ImageProps } from 'expo-image';

export const Button = styled(TouchableOpacity)`
  height: 330px;
  width: 130px;
`;

export const ImageCard = styled(Image).attrs<ImageProps>({
  contentFit: 'contain',
})`
  height: 100%;
  width: 100%;
`;
