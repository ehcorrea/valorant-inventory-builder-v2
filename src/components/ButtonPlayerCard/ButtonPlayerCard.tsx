import { TouchableOpacityProps } from 'react-native';

import * as S from './ButtonPlayerCard.styles';

export type ButtonPlayerCardProps = {
  image: string;
} & TouchableOpacityProps;

export function ButtonPlayerCard({ image, ...props }: ButtonPlayerCardProps) {
  return (
    <S.Button {...props}>
      <S.ImageCard source={image} />
    </S.Button>
  );
}
