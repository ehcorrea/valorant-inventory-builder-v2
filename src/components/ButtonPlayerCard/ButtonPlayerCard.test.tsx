import { fireEvent, render } from '@/test/utils';

import { ButtonPlayerCard } from './ButtonPlayerCard';

const mockedOnPress = jest.fn();

const setup = () => {
  return render(
    <ButtonPlayerCard
      onPress={mockedOnPress}
      image="placeholder-image.png"
      testID="playerCard-button"
    />
  );
};

describe('given that the component is rendered', () => {
  test('render image passed by props', () => {
    const container = setup();
    expect(
      container
        .getByTestId('playerCard-button')
        .findByProps({ source: 'placeholder-image.png' })
    ).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('pressing calls a callback', () => {
    const container = setup();
    const button = container.getByTestId('playerCard-button');
    fireEvent.press(button);
    expect(mockedOnPress).toHaveBeenCalled();
  });
});
