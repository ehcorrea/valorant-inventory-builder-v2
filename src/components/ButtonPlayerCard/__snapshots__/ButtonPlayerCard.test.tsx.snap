// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given that the component is rendered render image passed by props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 330,
      "opacity": 1,
      "width": 130,
    }
  }
  testID="playerCard-button"
>
  <Image
    contentFit="cover"
    source="placeholder-image.png"
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
